Library/Homebrew/manpages/brew.1.md.erb

```diff
@@ -144,6 +144,19 @@ Note that environment variables must have a value set to be detected. For
 example, run `export HOMEBREW_NO_INSECURE_REDIRECT=1` rather than just
 `export HOMEBREW_NO_INSECURE_REDIRECT`.
 
+`HOMEBREW_*` environment variables can also be set in Homebrew's environment
+files:
+
+* `/etc/homebrew/brew.env` (system-wide)
+
+* `$HOMEBREW_PREFIX/etc/homebrew/brew.env` (prefix-specific)
+
+* `$HOME/.homebrew/brew.env` (user-specific)
```

Comment:
Could we use `XDG_CONFIG_HOME` here? i.e.
```suggestion
* `$XDG_CONFIG_HOME/homebrew/brew.env` (user-specific)
```
with `XDG_CONFIG_HOME` defaulting to `$HOME/.config`. We already use `XDG_CACHE_HOME` for `HOMEBREW_CACHE` on Linux.

Reply:
@carlocab If it's set in the environment: yeh, I think so. If it's not: just use `$HOME`, right?

Reply:
We follow the XDG base directory [spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) for `XDG_CACHE_HOME`, which is to default to `$HOME/.cache` when unset. I think it will be simpler to do the same for `XDG_CONFIG_HOME`, which is to default to `$HOME/.config` when unset.

That said, not everybody follows the spec, and some tools do default to `$HOME` when unset, so it's not too out of the ordinary to not follow the spec.

Personally choosing well-known subdirectories of `$HOME` makes more sense, since that makes it easier to put the said subdirectory into a version-controlled dotfiles repo. Putting `$HOME` under version-control is a bit more annoying.

Reply:
I agree with following that spec on Linux, I don't think it makes sense to do on macOS unless set.

Reply:
@carlocab pushed in 375a7ee8d to use `XDG_CONFIG_HOME`.